File: python3/singly_link_list_node.py
class SinglyLinkListNode:
    next: 'SinglyLinkListNode'

    def __init__(self, value=None, next=None):
        self.value = value
        self.next = next

    def traverse(self):
        values = []
        node = self
        while node is not None:
            values.append(node.value)
            node = node.next
        return values

File: php/src/SinglyLinkListNode.php
<?php


namespace DataStructure;


class SinglyLinkListNode
{
    public function __construct(public $value = null, public ?SinglyLinkListNode $next = null)
    {
    }

    public function traverse(): array
    {
        $arr = [];

        if (! $this->value)
        {
            return $arr;
        }

        $arr[] = $this->value;

        if (! $this->next)
        {
            return $arr;
        }

        $values = $this->next->traverse();
        $arr = array_merge($arr, $values);

        return $arr;
    }

    public function remove(mixed $value)
    {

    }
}